client/client.go

```diff
@@ -5333,6 +5536,58 @@ func ParseAssociateClinicianToUserResponse(rsp *http.Response) (*AssociateClinic
 	return response, nil
 }
 
+// ParseListMembershipRestrictionsResponse parses an HTTP response from a ListMembershipRestrictionsWithResponse call
+func ParseListMembershipRestrictionsResponse(rsp *http.Response) (*ListMembershipRestrictionsResponse, error) {
+	bodyBytes, err := io.ReadAll(rsp.Body)
+	defer func() { _ = rsp.Body.Close() }()
```

Comment:
Is there a reason to put the `.Close` in a function and not defer it directly?
`defer rsp.Body.Close()`